src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -3587,49 +3541,43 @@ namespace HeatBalanceSurfaceManager {
-                                                                 state.dataConstruction->Construct(ConstrNumSh).AbsDiffBack(
-                                                                         IGlass);
-                            if (ShadeFlag == IntBlindOn || ShadeFlag == ExtBlindOn) {
+                            if (ANY_SHADE_SCREEN(ShadeFlag)) {
+                                SurfWinQRadSWwinAbs(IGlass, SurfNum) += QS(solEnclosureNum) * state.dataConstruction->Construct(ConstrNumSh).AbsDiffBack(IGlass);
+                            } else if (BITF_TEST_ANY(BITF(ShadeFlag), BITF(WinShadingType::IntBlind) | BITF(WinShadingType::ExtBlind))) {
                                 Real64 BlAbsDiffBk = InterpSlatAng(SurfWinSlatAngThisTS(SurfNum), SurfWinMovableSlats(SurfNum),
                                         state.dataConstruction->Construct(ConstrNumSh).BlAbsDiffBack(_, IGlass)); // Glass layer back diffuse solar absorptance when blind in place
                                 SurfWinQRadSWwinAbs(IGlass, SurfNum) += QS(solEnclosureNum) * BlAbsDiffBk;
                             }
                         }
-                        if (ShadeFlag == IntShadeOn)
-                            SurfWinIntLWAbsByShade(SurfNum) =
-                                    QL(radEnclosureNum) * state.dataConstruction->Construct(ConstrNumSh).ShadeAbsorpThermal *
-                                    TMULT(radEnclosureNum);
-                        if (ShadeFlag == IntBlindOn) {
+                        if (ShadeFlag == WinShadingType::IntShade) {
+                            SurfWinIntLWAbsByShade(SurfNum) = QL(radEnclosureNum) * state.dataConstruction->Construct(ConstrNumSh).ShadeAbsorpThermal * TMULT(radEnclosureNum);
+                        } else if (ShadeFlag == WinShadingType::IntBlind) {
                             Real64 EffBlEmiss = InterpSlatAng(SurfWinSlatAngThisTS(SurfNum), SurfWinMovableSlats(SurfNum),
                                                        SurfaceWindow(SurfNum).EffShBlindEmiss); // Blind emissivity (thermal absorptance) as part of glazing system
                             SurfWinIntLWAbsByShade(SurfNum) = QL(radEnclosureNum) * EffBlEmiss * TMULT(radEnclosureNum);
                         }
-                        if (ShadeFlag == IntShadeOn || ShadeFlag == ExtShadeOn || ShadeFlag == BGShadeOn ||
-                            ShadeFlag == ExtScreenOn)
-                            SurfWinIntSWAbsByShade(SurfNum) =
-                                    QS(solEnclosureNum) * state.dataConstruction->Construct(ConstrNumSh).AbsDiffBackShade;
-                        if (ShadeFlag == IntBlindOn || ShadeFlag == ExtBlindOn || ShadeFlag == BGBlindOn) {
+                        if (ANY_SHADE_SCREEN(ShadeFlag)){
+                            SurfWinIntSWAbsByShade(SurfNum) = QS(solEnclosureNum) * state.dataConstruction->Construct(ConstrNumSh).AbsDiffBackShade;
+                        } else if (ANY_BLIND(ShadeFlag)) {
```

Comment:
I had the same question. I think according to the current logic, we cannot have `shadingflag == IntBlindOn && shadingflag == ExtScreenOn` (`shadingflag == 1 && shadingflag == 4`), because there is only one variable `shadingflag` to indicate which type of shades/blinds/screen/bgshade is on, and the `on` types are exclusive. Maybe we should also consider changing this logic.

Reply:
Yes, of course. I was remembering the storm window flag, which can be independent of shading status.